test/complex.jl

```diff
@@ -1187,3 +1187,66 @@ end
+        @test gcd(complex(T(5), T(1)), complex(T(3), T(5))) === complex(T(1), T(1))
+        @test gcd(complex(T(5), T(3)), complex(T(2), T(8))) === complex(T(5), T(3))
+        @test gcd(complex(T(135), T(-14)), complex(T(155), T(34))) === complex(T(5), T(12))
+        @test gcd(complex(T(4), T(22)), complex(T(17), T(1))) === complex(T(1), T(3))
+        @test gcd(complex(T(7), T(1)), complex(T(5), T(3))) === complex(T(1), T(1))
+        @test gcd(complex(T(2), T(2)), complex(T(-1), T(1))) === complex(T(1), T(1))
+
+        @test gcd(complex(T(85), T(0)), complex(T(1), T(13))) === complex(T(7), T(6))
+        @test gcd(complex(T(0), T(0)), complex(T(1), T(13))) === complex(T(1), T(13))
+        @test gcd(complex(T(1), T(13)), complex(T(0), T(0))) === complex(T(1), T(13))
+        @test gcd(complex(T(1), T(3)), complex(T(3), T(9))) === complex(T(1), T(3))
+        @test gcd(complex(T(1), T(3)), complex(T(-9), T(3))) === complex(T(1), T(3))
+        @test gcd(complex(T(1), T(3)), complex(T(-3), T(-9))) === complex(T(1), T(3))
+        @test gcd(complex(T(1), T(3)), complex(T(9), T(-3))) === complex(T(1), T(3))
+        @test gcd(complex(T(-3), T(-9)), complex(T(1), T(3))) === complex(T(1), T(3))
+        @test gcd(complex(T(0), T(0)), complex(T(0), T(0))) === complex(T(0), T(0))
+
+        @test gcd(complex(T(3), T(0)), complex(T(5), T(0))) === complex(T(1), T(0))
+        @test gcd(complex(T(0), T(3)), complex(T(0), T(5))) === complex(T(1), T(0))
+        @test gcd(complex(T(-3), T(0)), complex(T(5), T(0))) === complex(T(1), T(0))
+        @test gcd(complex(T(0), T(-3)), complex(T(0), T(-5))) === complex(T(1), T(0))
+
+        let x = typemax(T)
+            @test gcd(complex(x, x), complex(T(1), T(0))) === complex(T(1), T(0))
+            @test gcd(complex(x, x), complex(T(-1), T(0))) === complex(T(1), T(0))
+            @test gcd(complex(x, x), complex(T(0), T(1))) === complex(T(1), T(0))
+            @test gcd(complex(x, x), complex(T(0), T(-1))) === complex(T(1), T(0))
+
+            @test gcd(complex(-x, -x), complex(T(1), T(0))) === complex(T(1), T(0))
+            @test gcd(complex(-x, -x), complex(T(-1), T(0))) === complex(T(1), T(0))
+            @test gcd(complex(-x, -x), complex(T(0), T(1))) === complex(T(1), T(0))
+            @test gcd(complex(-x, -x), complex(T(0), T(-1))) === complex(T(1), T(0))
+
+            @test_broken gcd(complex(x, T(0)), complex(T(2), T(0))) === complex(T(1), T(0))
+            @test_broken gcd(complex(x, x), complex(T(2), T(0))) === complex(T(1), T(1))
+        end
+        let x = typemin(T)
+            @test gcd(complex(x, x), complex(T(1), T(0))) === complex(T(1), T(0))
+            @test gcd(complex(x, x), complex(T(-1), T(0))) === complex(T(1), T(0))
+            @test gcd(complex(x, x), complex(T(0), T(1))) === complex(T(1), T(0))
+            @test gcd(complex(x, x), complex(T(0), T(-1))) === complex(T(1), T(0))
+        end
```

Comment:
Should test with typemax and typemin on the right as well. Especially typemin is the interesting one, because `-typemin(Int) == typemin(Int)`.